src/components/ListView.jsx

```diff
@@ -32,77 +19,36 @@ function ListView() {
       return;
     }
 
-    setLoading(false);
     const unsubscribe = onSnapshot(collection(db, localToken), (snapshot) => {
       const snapshotDocs = [];
       snapshot.forEach((doc) => {
         snapshotDocs.push({ ...doc.data(), id: doc.id });
       });
       setItems(snapshotDocs);
-      setLoading(true);
+      setLoading(false);
     });
     return () => {
       //Used to remove the snapshot listener when the component is unmounted/unsubscribed
       unsubscribe();
     };
   }, [localToken, navigate]);
 
-  //
-  const handleCheckboxChange = async (e) => {
-    const itemId = e.target.name;
-    //create time variable to save time when user checked the box as purchasedTime
-    const datePurchased = Date.now();
-    //if user want to uncheck the item it can be done and purchasedDate is set to null again
-    if (e.target.checked) {
-      setUpdateToDb(localToken, itemId, 'purchasedDate', datePurchased);
-    } else {
-      setUpdateToDb(localToken, itemId, 'purchasedDate', null);
-    }
-  };
-
-  //persists checked box for 24 hours
-  function within24hours(date) {
-    let timeCheck = false;
-    const gap = currentTime - date;
-    if (gap < hours24gap) {
-      timeCheck = true;
-    }
-    return timeCheck;
-  }
-
-  //controls page render to ensure items.length can evaluate after the items have been retrieved from firebase
-  if (loading) {
-    return (
-      <div className="m-20">
-        {!items.length ? (
-          <Welcome />
-        ) : (
-          <div>
-            <h1 className="text-center text-2xl my-12">Your Shopping List:</h1>
-            <ul className="grid grid-cols-2 justify-around">
-              {items.map((item, idx) => (
-                <li className="flex flex-col my-4" key={idx}>
-                  <div>
-                    {` Item Name: ${item.itemName}`}{' '}
-                    <input
-                      type="checkbox"
-                      checked={within24hours(item.purchasedDate)}
-                      onChange={(e) => handleCheckboxChange(e)}
-                      name={item.id}
-                      aria-label={item.itemName} //what do we want to call this ('item', 'item.itemName', 'purchased item' .....)
-                    />
-                  </div>
-                  <div>{` Frequency: ${item.frequency}`}</div>
-                </li>
-              ))}
-            </ul>
-          </div>
-        )}
-      </div>
-    );
-  } else {
-    return null;
-  }
+  //loading state controls page render to ensure items.length can evaluate after the items have been retrieved from firebase
+  return (
+    <>
+      {loading ? (
+        <img src="img/loading.gif" alt="loading" className="m-auto" />
+      ) : (
+        <div className="m-20">
+          {items.length ? (
+            <ListLayout items={items} localToken={localToken} />
```

Comment:
Love this refactor. So much easier to read the code and understand what's happening here. Despite what I said about ternary, it works pretty well here. No complaints about this.